src/test/test-fd-util.c

```diff
@@ -161,6 +162,9 @@ static void test_rearrange_stdio(void) {
         pid_t pid;
         int r;
 
+        if (detect_container() > 0)
```

Comment:
`detect_container` is a broad check. I don't think it should be used to skip the whole test just because it doesn't work in LXD containers especially when the source of the failure hasn't been found yet.